cmd/thanos/compact.go

```diff
@@ -458,6 +462,35 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.progressMetrics {
+		g.Add(func() error {
+			if err := sy.SyncMetas(context.Background()); err != nil {
+				return errors.Wrapf(err, "could not sync metas")
+			}
+			originalMetas := sy.Metas()
+
+			groups, err := grouper.Groups(originalMetas)
+			if err != nil {
+				return errors.Wrapf(err, "could not group original metadata")
+			}
+
+			ps := compact.NewDefaultPlanSim(reg, planner)
+			if err = ps.ProgressCalculate(context.Background(), groups); err != nil {
+				return errors.Wrapf(err, "could not simulate planning")
+			}
+
+			for _, meta := range originalMetas {
+				groupKey := compact.DefaultGroupKey(meta.Thanos)
+				ps.ProgressMetrics.NumberOfIterations.WithLabelValues(groupKey)
+				ps.ProgressMetrics.NumberOfBlocksToMerge.WithLabelValues(groupKey)
+			}
```

Comment:
These lines are used to initialize gauge metrics for each group. If you want to initialize it, you cannot do it after the calculation is done.